### PR TITLE
Problem: not able to specify commission rates in gentx

### PR DESCRIPTION
**Pull request by yihuang**
Closes: #72
Solution:
- support several optional fields in validator section

### CODE REVIEW

File: pystarport/cluster.py [hunk 1]
```diff
@@ -806,13 +813,24 @@ def create_account(cli, account, use_ledger=False):
...
         if "staked" in node:
+            optional_fields = [
+                "commission_max_change_rate",
+                "commission_max_rate",
+                "commission_rate",
+                "details",
+                "security_contact",
+            ]
+            extra_kwargs = {
+                name: node["name"] for name in optional_fields if name in node
```
Comment (yihuang) [inline pystarport/cluster.py:824]:
```suggestion
                name: node[name] for name in optional_fields if name in node
```

**Review by tomtau — APPROVED** [review]
thanks! so in yaml, it should just be e.g.:
```yaml
chaintest:
  validators:
    - coins: 10cro
      staked: 10cro
      commission_rate: 0
...
```

**Comment by yihuang** [comment]
> thanks! so in yaml, it should just be e.g.:
> 
> ```yaml
> chaintest:
>   validators:
>     - coins: 10cro
>       staked: 10cro
>       commission_rate: 0
> ...
> ```

You may need to specify as string, or I should do a conversion to string in the code.

File: pystarport/cluster.py [hunk 2]
```diff
@@ -806,13 +813,24 @@ def create_account(cli, account, use_ledger=False):
...
         if "staked" in node:
+            optional_fields = [
+                "commission_max_change_rate",
+                "commission_max_rate",
+                "commission_rate",
+                "details",
+                "security_contact",
+            ]
+            extra_kwargs = {
+                name: node[name] for name in optional_fields if name in node
```
Comment (yihuang) [inline pystarport/cluster.py:824]:
```suggestion
                name: str(node[name]) for name in optional_fields if name in node
```

**Comment by tomtau** [comment]
> You may need to specify as string, or I should do a conversion to string in the code.

string is fine -- the commission rates are decimals, so better to be passed in as strings instead of floats

**Comment by yihuang** [comment]
> > You may need to specify as string, or I should do a conversion to string in the code.
> 
> string is fine -- the commission rates are decimals, so better to be passed in as strings instead of floats

local test seems to work, although when I pass an invalid value like "1.2", it get ignored siliently.

**Pull request merged**